subsys/net/lib/websocket/websocket.c

```diff
@@ -344,7 +367,20 @@ int websocket_connect(int sock, struct websocket_request *wreq,
 	strncpy(key_accept + key_len, WS_MAGIC, olen);
 
 	/* This SHA-1 value is then checked when we receive the response */
-	mbedtls_sha1(key_accept, olen + key_len, sec_accept_key);
+#ifdef CONFIG_MBEDTLS_PSA_CRYPTO_CLIENT
+	psa_status = psa_hash_compute(PSA_ALG_SHA_1, (const uint8_t *)key_accept, olen + key_len,
+				      sec_accept_key, sizeof(sec_accept_key), &hash_length);
+	if (psa_status != PSA_SUCCESS) {
+		NET_DBG("[%p] Cannot calculate sha1 (%d)", ctx, psa_status);
+		goto out;
+	}
+#else
+	ret = mbedtls_sha1(key_accept, olen + key_len, sec_accept_key);
+	if (ret) {
+		NET_DBG("[%p] Cannot calculate sha1 (%d)", ctx, ret);
+		goto out;
```

Comment:
In all error cases, I think we should set `ret` value accordingly? `psa_hash_compute()` doesn't set it at all, `mbedtls_sha1` returns a value from a different error space (mbedtls errors, not errno).